packages/sdk/client/src/packlets/shell/shell-runtime.ts

```diff
@@ -0,0 +1,69 @@
+//
+// Copyright 2023 DXOS.org
+//
+
+import { Event } from '@dxos/async';
+import { PublicKey } from '@dxos/keys';
+import { schema } from '@dxos/protocols';
+import { OsAppService, ShellDisplay, ShellLayout } from '@dxos/protocols/proto/dxos/iframe';
+import { createProtoRpcPeer, ProtoRpcPeer } from '@dxos/rpc';
+import { createIFramePort } from '@dxos/rpc-tunnel';
+
+export const SHELL_CHANNEL = 'dxos:shell';
+
+export type ShellRuntimeOptions = {
+  channel: string;
+  timeout: number;
+};
+
+export class ShellRuntime {
+  readonly layoutUpdate = new Event<ShellLayout>();
+  private readonly _params: ShellRuntimeOptions;
+  private _appRpc?: ProtoRpcPeer<{ OsAppService: OsAppService }>;
+  private _layout = ShellLayout.DEFAULT;
+
+  constructor({ channel = SHELL_CHANNEL, timeout = 1000 }: Partial<ShellRuntimeOptions> = {}) {
+    this._params = { channel, timeout };
+  }
+
+  get layout() {
+    return this._layout;
+  }
+
+  async setDisplay(display: ShellDisplay) {
```

Comment:
In general group RPC verbs based on pairing up functionality (i.e., less granular); e.g., do we ever update display and space separately?

Reply:
yes they do, the display is used to toggle whether or not the shell is visible while the space is used to toggle the currently selected space. They may be called together (e.g., select this space then close the shell) but they may also be called separately (e.g., select the space but don't yet close the shell or close the shell without selecting a new space)